src/analyses/base.ml

```diff
@@ -1637,28 +1637,93 @@ struct
     in
     let inv_bin_float (a, b) fkind c op =
       let meet_bin a' b'  = FD.meet a a', FD.meet b b' in
-      let meet_com oi = (* commutative *)
-        meet_bin (oi c b) (oi c a) in
-      let meet_non oi oo = (* non-commutative *)
-        meet_bin (oi c b) (oo a c) in
       match op with
-      | PlusA  -> meet_com FD.sub
+      | PlusA  ->
+        (* A + B = C, \forall a \in A. a + b_min > pred c_min \land a + b_max < succ c_max
+            \land a + b_max > pred c_min \land a + b_min < succ c_max
+           \rightarrow A = [min(pred c_min - b_min, pred c_min - b_max), max(succ c_max - b_max, succ c_max - b_min)]
+        *)
+        let reverse_add v v' = (match FD.minimal c, FD.maximal c, FD.minimal v, FD.maximal v with 
+            | Some c_min, Some c_max, Some v_min, Some v_max when Float.is_finite (Float.pred c_min) && Float.is_finite (Float.succ c_max) -> 
+              let l = Float.min (Float.pred c_min -. v_min) (Float.pred c_min -. v_max) in 
+              let h =  Float.max (Float.succ c_max -. v_max) (Float.succ c_max -. v_min) in 
+              FD.of_interval (FD.precision c) (l, h)
+            | _ -> v') in
+        meet_bin (reverse_add b a) (reverse_add a b)
+      | MinusA ->
+        (* A - B = C \ forall a \in A. a - b_max > pred c_min \land a - b_min < succ c_max 
+            \land a - b_min > pred c_min \land a - b_max < succ c_max
+           \rightarrow A = [min(pred c_min + b_max, pred c_min + b_min), max(succ c_max + b_max, succ c_max + b_max)]
+        *)
+        let a' = (match FD.minimal c, FD.maximal c, FD.minimal b, FD.maximal b with 
+            | Some c_min, Some c_max, Some b_min, Some b_max when Float.is_finite (Float.pred c_min) && Float.is_finite (Float.succ c_max) -> 
+              let l = Float.min (Float.pred c_min +. b_max) (Float.pred c_min +. b_min) in 
+              let h =  Float.max (Float.succ c_max +. b_min) (Float.succ c_max +. b_max) in 
+              FD.of_interval (FD.precision c) (l, h)
+            | _ -> a) in
+        (* A - B = C \ forall b \in B. a_min - b > pred c_min \land a_max - b < succ c_max 
+            \land a_max - b > pred c_min \land a_min - b < succ c_max 
+           \rightarrow B = [min(a_max - succ c_max, a_min - succ c_max), max(a_min - pred c_min, a_max - pred c_min)]
+           \rightarrow B = [a_min - succ c_max, a_max - pred c_min]
```

Comment:
Also here you performed a similar simplification to what I mentioned, why here but not in the other cases?

Reply:
I have to admit, that this was inconsistent. I have now applied the suggested _simplifications_ for subtraction and addition